prov/usnic/src/usdf_domain.c

```diff
@@ -410,46 +336,23 @@ bool usdf_domain_checkname(uint32_t version, struct usd_device_attrs *dap,
 	bool valid;
 	int ret;
 
-	USDF_DBG("checking domain name: version=%d, domain name='%s'\n",
-		 version, hint);
-
-	if (version) {
-		valid = false;
-
-		ret = usdf_domain_getname(version, dap, &reference);
-		if (ret < 0)
-			return false;
+        /* If no hint was provided, then the check is happy/valid/succeeded */
+	if (!hint)
+		return true;
 
-		/* If the reference name exists, then this is version 1.4 or
-		 * greater.
-		 */
-		if (reference) {
-			if (hint) {
-				/* Case 1 */
-				valid = (strcmp(reference, hint) == 0);
-			} else {
-				/* Case 2 */
-				valid = true;
-			}
-		} else {
-			/* Case 3 & 4 */
-			valid = (hint == NULL);
-		}
+	USDF_DBG("checking domain name: domain name='%s'\n", hint);
 
-		if (!valid)
-			USDF_DBG("given hint %s does not match %s -- invalid\n",
-				 hint, reference);
+	ret = usdf_domain_getname(version, dap, &reference);
+	if (ret < 0)
+		return false;
 
-		free(reference);
-		return valid;
-	}
-
-	/* If hint is non-NULL then assume the version is 1.4 if not provided.
-	 */
-	if (hint)
-		return usdf_domain_checkname(FI_VERSION(1, 4), dap, hint);
+	valid = (strcmp(reference, hint) == 0);
+	if (!valid)
+	USDF_DBG("given hint %s does not match %s -- invalid\n",
+		hint, reference);
```

Comment:
No braces AND no indent?

Reply:
Ooops; indenting fail.  Fixed.

FWIW: I stuck with the prior Linux-kernel-like style (no braces for 1-line blocks, `!foo` for checking for NULL, etc.).  But I have no strong preference here -- I updated all the places you noted per your comments.